src/api/firebase.js

```diff
@@ -103,3 +105,72 @@ export async function deleteItem() {
 	 */
 }
 
+export function comparePurchaseUrgency(itemA, itemB) {
+	// Inactive items
+	const dateLastPurchaseToDateA = itemA.dateLastPurchased
+		? itemA.dateLastPurchased.toDate()
+		: itemA.dateCreated.toDate();
+	const dateLastPurchaseToDateB = itemB.dateLastPurchased
+		? itemB.dateLastPurchased.toDate()
+		: itemB.dateCreated.toDate();
+
+	// Overdue items
+	const overdueItemA = differenceInDays(
+		itemA.dateNextPurchased.toDate(),
+		Date.now(),
+	);
+	const overdueItemB = differenceInDays(
+		itemB.dateNextPurchased.toDate(),
+		Date.now(),
+	);
+
+	const daysSinceLastPurchaseA = getDaysBetweenDates(
+		Date.now(),
+		dateLastPurchaseToDateA,
+	);
+	const daysSinceLastPurchaseB = getDaysBetweenDates(
+		Date.now(),
+		dateLastPurchaseToDateB,
+	);
+
+	// Other categories of item
+	const daysUntilPurchaseA = getDaysBetweenDates(
+		itemA.dateNextPurchased.toDate(),
+		Date.now(),
+	);
+	const daysUntilPurchaseB = getDaysBetweenDates(
+		itemB.dateNextPurchased.toDate(),
+		Date.now(),
+	);
+
+	const daysSinceLastPurchaseBoth =
+		daysSinceLastPurchaseA >= 60 && daysSinceLastPurchaseB >= 60;
+
+	const overdueItemsBoth = overdueItemA < 0 && overdueItemB < 0;
+
+	if (daysSinceLastPurchaseBoth && overdueItemsBoth) {
+		// both items are overdue, sort alphabetically
+		if (itemA.name < itemB.name) return -1;
```

Comment:
You'd want to convert the name to the same  case (uppercase or lowercase) before comparing. You could also make this a function so that you don't have to rewrite it.